src/main/java/frc/robot/poseestimator/PoseEstimatorMath.java

```diff
@@ -0,0 +1,94 @@
+		}
+		return twist;
+	}
+
+	public static Twist2d updateChangeInAngle(Twist2d twist, Rotation2d currentGyroAngle, Rotation2d lastGyroAngle) {
+		double rotationDifference = currentGyroAngle.getRadians() - lastGyroAngle.getRadians();
+		double wrappedRotationDifference = MathUtil.angleModulus(rotationDifference);
+		return new Twist2d(twist.dx, twist.dy, wrappedRotationDifference);
+	}
+
+	public static double[] getCombinedStandardDeviations(double[] odometryStandardDeviations, double[] visionStandardDeviations) {
+		double[] combinedStandardDeviations = new double[3];
+		for (int i = 0; i < combinedStandardDeviations.length; i++) {
+			double odometryStandardDeviation = odometryStandardDeviations[i];
+			double visionStandardDeviation = visionStandardDeviations[i];
+			combinedStandardDeviations[i] = getCombinedStandardDeviations(odometryStandardDeviation, visionStandardDeviation);
+		}
+		return combinedStandardDeviations;
+	}
+
+	public static double getCombinedStandardDeviations(double odometryStandardDeviation, double visionStandardDeviation) {
+		if (odometryStandardDeviation == 0) {
+			return 0;
+		}
+		double squaredVisionStandardDeviation = visionStandardDeviation * visionStandardDeviation;
+		return odometryStandardDeviation
+			/ (odometryStandardDeviation + Math.sqrt(odometryStandardDeviation * squaredVisionStandardDeviation));
+	}
+
+	public static double[] multiplyArrays(double[] array1, double[] array2) {
+		double[] result = new double[Math.min(array1.length, array2.length)];
+		for (int i = 0; i < result.length; i++) {
+			result[i] = array1[i] * array2[i];
+		}
+		return result;
+	}
+
+	public static Transform2d
+		useKalmanOnTransform(VisionObservation observation, Pose2d currentPoseEstimation, double[] odometryStandardDeviations) {
+		double[] combinedStandardDeviations = PoseEstimatorMath
+			.getCombinedStandardDeviations(observation.standardDeviations(), odometryStandardDeviations);
+		Transform2d visionDifferenceFromOdometry = new Transform2d(currentPoseEstimation, observation.visionPose());
+		return scaleDifferenceFromKalman(visionDifferenceFromOdometry, combinedStandardDeviations);
+	}
+
+	public static Transform2d
+		scaleDifferenceFromKalman(Transform2d visionDifferenceFromOdometry, double[] combinedStandardDeviations) {
+		double[] visionDifferenceFromOdometryMatrix = {
+			visionDifferenceFromOdometry.getX(),
+			visionDifferenceFromOdometry.getY(),
+			visionDifferenceFromOdometry.getRotation().getRadians()};
+		double[] standardDeviationsAppliedTransform = multiplyArrays(
+			combinedStandardDeviations,
+			visionDifferenceFromOdometryMatrix
+		);
+		return new Transform2d(
+			standardDeviationsAppliedTransform[PoseArrayEntryValue.X_VALUE.getEntryValue()],
+			standardDeviationsAppliedTransform[PoseArrayEntryValue.Y_VALUE.getEntryValue()],
+			Rotation2d.fromRadians(standardDeviationsAppliedTransform[PoseArrayEntryValue.ROTATION_VALUE.getEntryValue()])
+		);
+	}
```

Comment:
You're reducing the change in each axis in proportion to the so-called stdDev. This is *not* a kalman filter. wpilib [has it's own kalman filter](https://first.wpi.edu/wpilib/allwpilib/docs/development/java/edu/wpi/first/math/estimator/KalmanFilter.html) -- I think we should use it. 

(for context, kalman filter takes the input from the vision, and the observed pose from the odometry, and combines them into one pose, correspondingly to the estimated noise for each system)

Reply:
you are very much CORRECT! my usage of the stdevs is incorrect, I used a weighted average which is ok but it means that the stdevs are not really stdevs. I do want to rewrite everything and use the Kalman filter correctly. I think I would just make another PoseEstimator instead of rewriting the existing one